### PR TITLE
[UI] add union-types for candyShop props in activity

### PR DESCRIPTION
**Pull request by neversaynever0502**

### CODE REVIEW

File: core/ui/src/public/Activity/index.tsx [hunk 1]
```diff
@@ -18,9 +18,19 @@ dayjs.extend(relativeTime);
 import './style.less';
 import { useSocket } from 'public/Context/Socket';
 import { useUpdateCandyShopContext } from 'public/Context/CandyShopDataValidator';
+import { web3 } from '@project-serum/anchor';
+
+interface ShopInfo {
```
Comment (vanminh1701) [inline core/ui/src/public/Activity/index.tsx:23]: why we need add this interface ?

File: core/ui/src/public/Activity/index.tsx [hunk 2]
```diff
@@ -39,8 +49,9 @@ export const Activity: React.FC<ActivityProps> = ({ candyShop, identifiers, orde
 
   const getTrades = useCallback(
     (offset: number, limit: number, firstLoad?: boolean) => () => {
-      candyShop
-        .transactions({ identifiers, offset, limit, sortBy: orderBy })
+      
+      fetchTradeByShopAddress(candyShop.candyShopAddress, { identifiers, offset, limit, sortBy: orderBy })
+        // candyShop.transactions({ identifiers, offset, limit, sortBy: orderBy })
```
Comment (vanminh1701) [inline core/ui/src/public/Activity/index.tsx:54]: should we remove this comment ?

File: core/ui/src/public/Activity/index.tsx (hunk 1 above)
Comment (neversaynever0502) [reply core/ui/src/public/Activity/index.tsx:23]: Currently, the main UI components: Order/Auction/Activity in public all need candyShop: CandyShop as prop but the methods they called from sdk now can use without the CandyShop instance but just candyShopAddress or partial infos. 
So I created the new interface to define the necessary types about CandyShop used in this file.

File: core/ui/src/public/Activity/index.tsx (hunk 2 above)
Comment (neversaynever0502) [reply core/ui/src/public/Activity/index.tsx:54]: Yes, I forgot to remove it. 
I will remove it immediately.

File: core/ui/src/public/Activity/index.tsx [hunk 3]
```diff
@@ -50,7 +59,6 @@ export const Activity: React.FC<ActivityProps> = ({ candyShop, identifiers, orde
           if (hasMore) {
             setOffset(offset + count + 1);
           }
-
```
[the remark this answers is not in the document]
Comment (neversaynever0502) [reply core/ui/src/public/Activity/index.tsx:53]: OK, I already fixed it, thanks.

File: core/ui/src/public/Activity/index.tsx [hunk 4]
```diff
@@ -18,9 +18,19 @@ dayjs.extend(relativeTime);
...
+  env: web3.Cluster;
+  baseUnitsPerCurrency: number;
+  priceDecimalsMin: number;
+  priceDecimals: number;
+  explorerLink: ExplorerLinkBase;
+}
 
 interface ActivityProps {
-  candyShop: CandyShop;
+  candyShop: CandyShop | ShopInfo;
```
[the remark this answers is not in the document]
Comment (neversaynever0502) [reply core/ui/src/public/Activity/index.tsx:33]: @haVincy 
I tested with example, it worked.(as image)
Should I push the example code? 

image

[the remark this answers is not in the document]
Comment (haVincy) [reply core/ui/src/public/Activity/index.tsx:33]: @neversaynever0502 we can comment out that new add usage for `Activity` component for example to show it, keep the original usage live. 

```tsx
<!-- Can serve Activity with partial shop info without CandyShop instance to present the same --> 
<!--
put example usage inside the comment
-->
```

File: example/MarketplaceExample.tsx [hunk 5]
```diff
@@ -54,6 +54,15 @@ export const MarketplaceExample: React.FC<MarketplaceExampleProps> = ({ candySho
...
       <h1 style={{ textAlign: 'center', fontWeight: 'bold', margin: '80px 0 30px' }}>Activity</h1>
       <Activity candyShop={candyShop} orderBy={ORDER_ACTIVITY} />
+      <Activity candyShop={{
+        candyShopAddress: candyShop.candyShopAddress,
+        env: candyShop.env,
+        baseUnitsPerCurrency: candyShop.baseUnitsPerCurrency,
+        priceDecimalsMin: candyShop.priceDecimalsMin,
+        priceDecimals: candyShop.priceDecimals,
+        explorerLink: candyShop.explorerLink,
+      }} orderBy={ORDER_ACTIVITY} />
```
Comment (haVincy) [inline example/MarketplaceExample.tsx:64]: Like I said above, we should comment this out with description.

Comment (neversaynever0502) [reply example/MarketplaceExample.tsx]: I have added description and JSX comment  {/* */} .
please check again, thanks.

image

**Review by haVincy — APPROVED** [review]
LGTM!

**Pull request merged**